source/particle/world.cc

```diff
@@ -1606,18 +1644,32 @@ namespace aspect
       generator->parse_parameters(prm);
       generator->initialize();
 
-      // Create an output object depending on what the parameters specify
-      output.reset(Output::create_particle_output<dim>
-                   (prm));
-
-      // We allow to not generate any output plugin, in which case output is
-      // a null pointer. Only initialize output if it was created.
-      if (output)
+      // Remove duplicate output formats specified in the parameter file.
+      output_format_names.erase(std::unique(output_format_names.begin(), output_format_names.end()),
+                                output_format_names.end());
+
+      // Create an output object depending on what the parameters specify so long as none has not been specified.
+      // If none and more than one other format is specified, then we ignore none.
+      AssertThrow ((std::find (output_format_names.begin(),
+                               output_format_names.end(),
+                               "none") == output_format_names.end())
+                   ||
+                   (output_format_names.size() == 1),
+                   ExcMessage ("If you specify 'none' for the output format for parameter \"Data output format\","
+                               "then this needs to be the only value given."));
+
+      if (output_format_names.size() != 1 || output_format_names[0] != "none")
```

Comment:
After the AssertThrow above we are sure that if "none" was selected the output_format_names vector has size 1. So you could add the following line:
```
if (output_format_names[0] == "none")
  output_format_names.clear();
```
This would allow you to simplify the logic in the current line to `if (output_format_names.size() > 0)` and include the other loop below into this if-loop. I feel that would be easier to understand.